src/lm/sentence.cpp

```diff
@@ -0,0 +1,177 @@
+/**
+ * @file sentence.cpp
+ * @author Sean Massung
+ */
+
+#include <iostream>
+#include <sstream>
+#include <numeric>
+#include <iterator>
+#include "lm/sentence.h"
+#include "analyzers/analyzer.h"
+#include "analyzers/tokenizers/icu_tokenizer.h"
+#include "analyzers/tokenizers/whitespace_tokenizer.h"
+#include "analyzers/filters/all.h"
+
+namespace meta
+{
+namespace lm
+{
+sentence::sentence(const std::string& text, bool tokenize /* = true */)
+{
+    if (tokenize)
+    {
+        using namespace analyzers;
+        std::unique_ptr<token_stream> stream;
+        stream = make_unique<tokenizers::icu_tokenizer>();
+        stream = make_unique<filters::empty_sentence_filter>(std::move(stream));
+        std::string text_copy{text}; // consider changing parameter to non-const
+        stream->set_content(std::move(text_copy));
+        while (*stream)
+            tokens_.push_back(stream->next());
```

Comment:
I think you'd be better off just using `utf::segmenter` directly here instead of going through the analyzer pipeline interface. This will eliminate the unnecessary copy of the `std::string` parameter, since `utf::segmenter` just takes a `util::string_view`, and I think the code will be simpler.